### PR TITLE
WIP: Test Indexer boxes functionality

### PR DESCRIPTION
**Pull request by algochoi**
This PR adds Indexer path tests based on this Indexer branch that supports box search: https://github.com/algorand/indexer/pull/1117

Sample Python SDK implementation: https://github.com/algorand/py-algorand-sdk/pull/366

#176

### CODE REVIEW

**Review by ahangsu — APPROVED** [review]

**Comment by michaeldiamant** [comment]
For reviewer convenience, here's the related SDK PRs:
* https://github.com/algorand/go-algorand-sdk/pull/355
* https://github.com/algorand/java-algorand-sdk/pull/356
* https://github.com/algorand/js-algorand-sdk/pull/619
* https://github.com/algorand/py-algorand-sdk/pull/366

File: features/unit/v2indexerclient_paths.feature [hunk 1]
```diff
@@ -183,6 +183,29 @@ Feature: Indexer Client v2 Paths
...
+  @unit.applications.boxes
+  Scenario Outline: SearchForApplicationBoxes path
+    When we make a SearchForApplicationBoxes call with applicationID <application-id> with max <max> nextToken "<nextToken>"
+    Then expect the path used to be "<path>"
+
+    Examples:
+      | path                                           | application-id | max | nextToken |
+      | /v2/applications/1234/boxes                    | 1234           | 0   |           |
+      | /v2/applications/1234/boxes?limit=2            | 1234           | 2   |           |
+      | /v2/applications/1234/boxes?limit=2&next=token | 1234           | 2   | token     |
```
Comment (tzaffi) [inline features/unit/v2indexerclient_paths.feature:207]: If we assume that the token is passed directly back (which is my opinion) then this works well. If we assume that the the SDK tries to parse `nextToken` and then reproduce it `b64` encoded, then more complex unit tests would be called for. Cases such as:
1. `nextToken=b64:bmV4dEJveA==` produces `/v2/applications/1234/boxes?limit=2&next=b64:bmV4dEJveA==`
2. `nextToken=str:nextBox` produces `/v2/applications/1234/boxes?limit=2&next=b64:bmV4dEJveA==`
But again, I don't favor this more complex approach.

Comment (tzaffi) [reply features/unit/v2indexerclient_paths.feature:207]: The previous comment is misleading. Let me try again. Let's look at [this autogenerated fixture](https://github.com/algorand/indexer/blob/98df20f7a0fb1c515f2faa25de7579aa429234d3/api/test_resources/boxes.json#L764).

In particular, the request url is:
```url
http://localhost:8999/v2/applications/1/boxes?limit=3\u0026next=b64%3AYm94ICM4
```
and the response was

```json
{
    "statusCode": 200,
    "body": "{\"application-id\":1,\"boxes\":[{\"name\":\"ZGlzYXBwb2ludGluZyBib3g=\"},{\"name\":\"ZG9uJ3QgYm94IG1lIGluIHRoaXMgd2F5\"},{\"name\":\"ZmFudGFidWxvdXM=\"}],\"next-token\":\"ZmFudGFidWxvdXM=\"}\n"
},
```
So really what this example is stating is that if the _most recently received_ token was `token`, we should therefore supply `&next=b64:token` (and that is strange because `token` is not a valid b64 encoding)

Comment (tzaffi) [reply features/unit/v2indexerclient_paths.feature:207]: The previous comment is misleading. Let me try again. Let's look at [this autogenerated fixture](https://github.com/algorand/indexer/blob/98df20f7a0fb1c515f2faa25de7579aa429234d3/api/test_resources/boxes.json#L764).

In particular, the request url is:
```url
http://localhost:8999/v2/applications/1/boxes?limit=3\u0026next=b64%3AYm94ICM4
```
and the response was

```json
{
    "statusCode": 200,
    "body": "{\"application-id\":1,\"boxes\":[{\"name\":\"ZGlzYXBwb2ludGluZyBib3g=\"},{\"name\":\"ZG9uJ3QgYm94IG1lIGluIHRoaXMgd2F5\"},{\"name\":\"ZmFudGFidWxvdXM=\"}],\"next-token\":\"ZmFudGFidWxvdXM=\"}\n"
},
```
So really what this example is stating is that if the _most recently received_ token was `token`, we should therefore supply `&next=b64:token` (and that is strange becose `token` is not a valid b64 encoding)


More context in [the swagger](https://github.com/algorand/indexer/blob/3457a5ace6d56f2b76584335547c73729e889f96/api/indexer.oas2.json#L3048)

Comment (ahangsu) [reply features/unit/v2indexerclient_paths.feature:207]: Hey, from our previous discussion, I remembered we concluded that we would accept b64 encoded `next` token, and I think the testcases on `next` token should be modified?

Comment (tzaffi) [reply features/unit/v2indexerclient_paths.feature:207]: am resolving and gonna put a new comment entirely in place of all these stale statements

File: features/unit/v2indexerclient_paths.feature [hunk 2]
```diff
@@ -183,6 +183,29 @@ Feature: Indexer Client v2 Paths
...
+
+  @unit.applications.boxes
+  Scenario Outline: SearchForApplicationBoxes path
+    When we make a SearchForApplicationBoxes call with applicationID <application-id> with max <max> nextToken "<nextToken>"
+    Then expect the path used to be "<path>"
+
+    Examples:
+      | path                                           | application-id | max | nextToken |
+      | /v2/applications/1234/boxes                    | 1234           | 0   |           |
+      | /v2/applications/1234/boxes?limit=2            | 1234           | 2   |           |
```
Comment (tzaffi) [inline features/unit/v2indexerclient_paths.feature:206]: For the purposes of a unit test, the example on line 207 is fine. However, in practice we expect `nextToken` to be of the following format `b64:B64_ENCODED_STRING`. Consequently, a slightly better test would follow the same format. e.g.
```
 | /v2/applications/1234/boxes? Limit=2&next=b64:dG9rZW4= | 1234           | 2   | b64:dG9rZW4=    |
```

Comment (tzaffi) [reply features/unit/v2indexerclient_paths.feature:206]: (or the url-encoded version) I'm not entirely sure why these examples are not url-encoded ????

Comment (ahangsu) [reply features/unit/v2indexerclient_paths.feature]: just changed the testcase accordingly.

**Review by tzaffi — APPROVED** [review]
